bazel/repository.bzl

```diff
@@ -7,9 +7,13 @@ def opentelemetry_cpp_deps():
     # Google Benchmark library.
     # Only needed for benchmarks, not to build the OpenTelemetry library.
     maybe(
-        native.local_repository,
+        http_archive,
         name = "com_github_google_benchmark",
-        path = "third_party/benchmark",
+        sha256 = "dccbdab796baa1043f04982147e67bb6e118fe610da2c65f88912d73987e700c",
+        strip_prefix = "benchmark-1.5.2",
+        urls = [
+            "https://github.com/google/benchmark/archive/v1.5.2.tar.gz",
+        ],
```

Comment:
Thanks for the fix. Just wondering if it would be better to directly upgrade the submodule ?

Reply:
The problem is that with submodule is hard to use with bazel, because I need to redefine the repository, with http archive this is no longer a problem, and the most annoying is the proto because it does not work without that.

Reply:
When I say use is depending on this project from a third party

Reply:
To clarify, right now I have to do this:

```
# Load OpentTelemetry-CPP dependency
http_archive(
    name = "com_github_opentelemetry_cpp",
    strip_prefix = "opentelemetry-cpp-7d7bd5247f0b546cb674afd82a6ef57be91a83d1",
    sha256 = "a2940df54fb0087715f96e4cb6c860987a3141100fb27fcf0871fe2638c82f7e",
    urls = [
        "https://github.com/open-telemetry/opentelemetry-cpp/archive/7d7bd5247f0b546cb674afd82a6ef57be91a83d1.tar.gz",
    ],
)

# Redefine the com_github_opentelemetry_proto because uses submodule
http_archive(
        name = "com_github_opentelemetry_proto",
        build_file = "@com_github_opentelemetry_cpp//bazel:opentelemetry_proto.BUILD",
        sha256 = "08f090570e0a112bfae276ba37e9c45bf724b64d902a7a001db33123b840ebd6",
        strip_prefix = "opentelemetry-proto-0.6.0",
        urls = [
            "https://github.com/open-telemetry/opentelemetry-proto/archive/v0.6.0.tar.gz",
        ],
)
```

So if we use always http_archive for all the bazel repos I don't have to do the redefinition.

Reply:
https://github.com/open-telemetry/opentelemetry-cpp/issues/461

Reply:
Thanks for explaining, that makes sense to me now. This would be the issue for all the libraries bazel picks from submodule. Will look forward for comments from @jsuereth our bazel expert :)

Reply:
Done.